power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp

```diff
@@ -364,6 +367,18 @@ class Deserializer {
     using DataByteMeta = std::vector<std::vector<ComponentByteMeta>>;
     using AttributeByteMeta = std::vector<std::pair<std::string_view, std::vector<std::string_view>>>;
 
+    using Buffer = WritableDataset::Buffer;
+    struct BufferView {
+        Buffer const* buffer{nullptr};
+        Idx idx{0};
+        std::vector<AttributeBuffer<void>> reordered_attribute_buffers;
```

Comment:
```suggestion
        std::span<AttributeBuffer<void> const> reordered_attribute_buffers;
```

Maybe make them a span and store the owned copy in `parse_component` which outlives the whole parsing of the component.

Reply:
why? this is very closely related to the actual buffer. After the buffer is gone, there's no point in keeping this mapping. The only real reason I can think of is that the serializer (not the deserializer) does things scenario-by-scenario, and this would keep the similarity between the two. however, there are already so many differences between the serializer and deserializer that i'm not sure whether this will make it more readable.

Reply:
See my other comment 

https://github.com/PowerGridModel/power-grid-model/pull/708#discussion_r1743568744

Previously I though this `vector` will not be copied around so the memory and CPU overhead is negligible. However, in your `advance` function you implicitly copy the `BufferView`. And `advance` is called for every single element and scenario.

Reply:
EDIT: apologies, i missed your other comment. Yes, this is a good solution for the case that there are very many scenarios with few components (e.g. N-1)

Reply:
i'd propose keeping it next to `attributes`, which does not outlive `parse_component` but has the same lifetime

Reply:
You also call `advance` in all elements. So the number of copy operations of `BufferView` is the same as the number of scenarios times number of elements per scenario in one component type.

Reply:
> i'd propose keeping it next to `attributes`, which does not outlive `parse_component` but has the same lifetime

That's should be a good proposal.